dvc/stage.py

```diff
@@ -599,6 +599,8 @@ def create(repo, accompany_outs=False, **kwargs):
                 raise StageFileAlreadyExistsError(stage.relpath)
 
             os.unlink(path)
+            # Removing stage file makes `repo.stages` invalid, need to reset.
+            repo.reset()
```

Comment:
I don't think we should do it here. We should `repo._reset` in things like `repo.add/run/imp/etc`. I know that we are using `reset` in the wrapper above, but that was justified and this one feels like spreading the context too much.

Reply:
Or am I missing something?

Reply:
I agree that it does not feel well to put it here, I will try to find another solution.